File: 4_Semestre/Software Básico/Pilha/multiplos.c
#include <stdio.h>

long int v = 10;
long int t = 5;

long int multiplos() {
    long int m = 1;
    for (long int i = 2; i <= t; i++)
        if (v % i == 0)
            m += 1;
    return m;
}

int main() {
    return multiplos();
}

File: Paradigmas de Programacao/14Semana/multiplos.hs
module Praticando where

multiplos :: Int -> [Int]
multiplos n = take 10 [x | x <- [1, 2..], mod x n == 0]

multiplos2 :: Int -> [Int]
multiplos2 n = [n*x | x <- [1..10]]

primo :: Int -> Bool
primo n = if length [x | x <- [1..n], mod n x == 0] <= 2
  then True
  else False
